compiler/rustc_hir_analysis/src/check/compare_impl_item.rs

```diff
@@ -1886,8 +1886,6 @@ fn compare_type_predicate_entailment<'tcx>(
     let impl_ty_predicates = tcx.predicates_of(impl_ty.def_id);
     let trait_ty_predicates = tcx.predicates_of(trait_ty.def_id);
 
-    check_region_bounds_on_impl_item(tcx, impl_ty, trait_ty, false)?;
```

Comment:
There was no need to move this one out of `compare_type_predicate_entailment` except consistency with both `compare_{method,const}_predicate_entailment` which is reason enough imo.